lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6176,6 +6176,56 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
            "Functions without returns must have been diagnosed");
     auto *origExit = &*origExitIt;
 
+    SmallVector<SILValue, 8> origFormalResults;
+    collectAllFormalResultsInTypeOrder(original, origFormalResults);
+    auto origResult = origFormalResults[getIndices().source];
+
+    // If original result is non-varied, it will always have a zero derivative.
+    // Skip full pullback generation and simply return zero for wrt parameters.
+    //
+    // NOTE(TF-876): This shortcut is currently necessary for functions
+    // returning non-varied result with >1 basic block where some basic blocks
+    // have no dominated active values; control flow differentiation does not
+    // handle this case. See TF-876 for context.
+    if (!getActivityInfo().isVaried(origResult, getIndices().parameters)) {
```

Comment:
For modularity, please refactor the body of this `if` block to a separate method.

Reply:
Done in https://github.com/apple/swift/pull/27554/commits/09b3877c8238ad45a7402548b15074ab1f55b368.